File: src/jbased/Part6_2.java
package jbased;

import org.nd4j.linalg.api.ndarray.INDArray;
import org.nd4j.linalg.factory.Nd4j;
import org.nd4j.linalg.ops.transforms.Transforms;
public class Part6_2 {
    public static void main(String[] args) {
        double[][] layerOutputs = {{4.8, 1.21, 2.385},
                {8.9, -1.81, 0.2},
                {1.41, 1.051, 0.026}};
        INDArray nd1 = Nd4j.create(layerOutputs);
        nd1 = Transforms.exp(nd1);
        INDArray ndSum = nd1.sum(1);
        INDArray result = nd1.divColumnVector(ndSum);
        ndSum = result.sum(1);
        System.out.println(result.toStringFull());
        System.out.println(ndSum.toStringFull());
    }
}

File: src/kbased/Part6_2.kt
package kbased

import org.nd4j.linalg.factory.Nd4j
import org.nd4j.linalg.ops.transforms.Transforms

fun main(args: Array<String>) {
    val layerOutputs = arrayOf(doubleArrayOf(4.8, 1.21, 2.385), doubleArrayOf(8.9, -1.81, 0.2), doubleArrayOf(1.41, 1.051, 0.026))
    var nd1 = Nd4j.create(layerOutputs)
    nd1 = Transforms.exp(nd1)
    var ndSum = nd1.sum(1)
    val result = nd1.divColumnVector(ndSum)
    ndSum = result.sum(1)
    println(result.toStringFull())
    println(ndSum.toStringFull())
    }